app/models/accounting/payment.rb

```diff
@@ -23,25 +25,27 @@ class Payment < ::ActiveRecord::Base
 
     before_validation :format_data
 
-    validates :address, presence: true
+    validates :address, presence: true, if: :card?
 
     validates :account_type, inclusion: { in: Accounting::Payment.account_types.keys }, if: :ach?
 
-    validates :number, length: { in: 13..16 }, if: :card?
+    validates :number, length: { in: 13..16 }, if: :card?, unless: :accept?
+
+    validates :number, :ccv, presence: true, if: :card?, unless: :accept?
 
-    validates :number, :ccv, :month, :year, presence: true, if: :card?
+    validates :number, :ccv, numericality: { only_integer: true }, if: :card?, unless: :accept?
```

Comment:
@ninjarails - I don't think we should still have fields for the number or the ccv.  I could see the number, if it's only the last four of the card.  I don't know why we'd ever have ccv.  Aren't we just getting the token or sconce back from Authorize to store?

Reply:
I don't actually use some of these validations for Accept.js. But it's used when CSR creates payment method.